Tests/MongoSwiftTests/MongoCollectionTests.swift

```diff
@@ -110,6 +110,25 @@ final class MongoCollectionTests: MongoSwiftTestCase {
         expect(try self.coll.insertOne(self.doc1)).toNot(throwError())
     }
 
+    func testDropWithWriteConcerns() throws {
+        let encoder = BSONEncoder()
+
+        let center = NotificationCenter.default
+
+        let writeConcern = try WriteConcern(journal: true, w: .number(1))
+        let opts = DropCollectionOptions(writeConcern: writeConcern)
+
+        let observer = center.addObserver(forName: nil, object: nil, queue: nil) { notif in
+            print(notif)
+        }
+
+        expect(try self.coll.drop(options: opts)).toNot(throwError())
+        // insert something so we don't error when trying to drop again in teardown
+        expect(try self.coll.insertOne(self.doc1)).toNot(throwError())
+
+        center.removeObserver(observer)
```

Comment:
rather than removing the observer here, we should put this in a `defer` statement right below where we create it, like `defer { center.removeObserver(observer) }`. 

this ensures that if the function doesn't finish running (for ex. that could happen if one of the lines between adding the observer and here throws an error) that we will still remove the observer when the function exits.

you can read a little more about defer statements [here](https://docs.swift.org/swift-book/ReferenceManual/Statements.html) (scroll down). basically they are useful for cleanup code like this 🙂